lib/req/request.ex

```diff
@@ -632,6 +653,7 @@ defmodule Req.Request do
         inspect: &IO.inspect/1
       )
   """
+  @spec append_request_steps(t(), keyword(fun())) :: t()
```

Comment:
I think we might want to make the `request_step` and friends *public* types, as they are documented in the module doc anyway. Thoughts?

Reply:
Yeah we can add the type and use it in specs. I’d hold off on speccing request.request_specs though, I want to have an option to add metadata around steps in the future if I need to.

Reply:
Ah nevermind!